src/main/java/org/synyx/urlaubsverwaltung/application/application/ApplicationInteractionService.java

```diff
@@ -64,6 +64,16 @@ public interface ApplicationInteractionService {
      */
     Application cancel(Application application, Person canceller, Optional<String> comment);
 
+    /**
+     * Sets the application's state to cancelled without an approval.
+     *
+     * @param application to cancel directly
+     * @param canceller   executes the application's cancellation
+     * @param comment     giving further information to cancellation of application for leave (is optional)
+     * @return the cancelled application for leave
+     */
+    Application directCancel(Application application, Person canceller, Optional<String> comment);
```

Comment:
Da die Art des Stornieren von der Application abhängt, sehe ich das in der Implementierung von `cancel`. Schnittstelle klein/einfach halten und gar nicht erst die Option bieten ein `directCancel( application(requiresApproval=true) )` aufrufen zu können. Diesen Fall sollte vielleicht sogar noch abgefangen werden in `directCancel`, falls es einen Grund gibt die Methode zu behalten.

Reply:
ah, okay. bei `allow` / `directAllow` ist es auch so 🤔